### ISSUE TITLE
Narrowing doesn't occur with ternary if

### ISSUE THREAD

**Issue by dr-carlos** [astral-sh/ty#2804]
### Summary

It seems that ty doesn't narrow from an `... if ... else ...` (ternary if expression) where it would with an if-else statement.

```python
from random import choice
from typing import assert_type, Literal

def func() -> bool: return choice((True, False))

def works(flag: bool) -> None:
    var = func()

    if flag:
        if var:
            assert_type(var, Literal[True])
    else:
        if var:
            assert_type(var, Literal[True])

def fails(flag: bool) -> None:
    var = func()

    if (
        var
        if flag
        else
        var
    ):
        assert_type(var, Literal[True])
```

https://play.ty.dev/46ca4217-a005-43d2-a494-b59862563755

### Version

ty 0.0.16

**Comment by sharkdp** [comment]
To simplify a bit, I understand that you want this to reveal `Literal[True]` instead of `bool`?

```py
def _(var: bool, flag: bool) -> None:
    if var if flag else var:
        reveal_type(var)
```

This is not supported by any major typechecker (pyright, mypy, pyrefly). While it seems feasible in principle, this looks like a rather peculiar pattern. Maybe you can share the actual real world code? But it seems unlikely to me that we're going to support this in ty.

**Comment by dr-carlos** [comment]
Thanks!

Sorry for not checking with another type checker. It seemed like an obvious inclusion in my head (anecdotally, ty seems to narrow based on `and` or `or`, but not a ternary if), but I should have checked.

Here's a snapshot of the more complex real-world code: https://play.ty.dev/c5ad0a93-5bfd-4bc6-94df-b6474cf260e5

And the relevant section of that snapshot:

```python
if not (
	(
		isinstance(
			tag := event.find(attr.value),
			Tag,
		)
		and isinstance(value := tag.attrs.get("src"), str)
	)
	if attr is EventAttribute.IMAGE_NAME
	else (
		isinstance(
			tag := event.find(id=attr.value),
			Tag,
		)
		and isinstance(value := tag.attrs.get("value"), str)
	)
):
	return None

assert_type(tag, Tag)
assert_type(value, str)
```

It's a bit messy, but it's very similar code for both branches, but different enough that I've used a ternary if. And if either the `tag` or `value` are `None`, the function will return. So `ty` should be able to safely narrow both variables.
